src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/BuildingRequestResolver.java

```diff
@@ -129,6 +111,24 @@ public void resolve(
         manager.updateRequestState(request.getToken(), RequestState.COMPLETED);
     }
 
+    @NotNull
+    private AbstractBuilding getBuildingFromRequest(
+      @NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request) throws RuntimeException
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00112)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.RuntimeException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)